src/vs/workbench/parts/debug/browser/debugActions.ts

```diff
@@ -230,7 +235,20 @@ export class RestartAction extends AbstractDebugAction {
 			process = this.debugService.getViewModel().focusedProcess;
 		}
 		if (!process) {
-			return TPromise.as(null);
+			const configurationManager = this.debugService.getConfigurationManager();
+			let launch = configurationManager.selectedConfiguration.launch;
+			if (!launch) {
+				const rootUri = this.historyService.getLastActiveWorkspaceRoot();
```

Comment:
All of this code is now duplicated. This is not good.
Either create a helper function which will be used inside both RestarAction and StartAction, or RestartAction should create a StartAction as you previously have done it and I wrongly corrected you

Reply:
Can you please adress this?

Reply:
Will do - I've been busy the past couple of days.